controllers/apps/transformer_component_rbac.go

```diff
@@ -57,145 +61,189 @@ func (t *componentRBACTransformer) Transform(ctx graph.TransformContext, dag *gr
 		return nil
 	}
 
-	graphCli, _ := transCtx.Client.(model.GraphClient)
-
-	serviceAccount, err := buildServiceAccount(transCtx)
-	if err != nil {
-		return err
+	var serviceAccountName string
+	var sa *corev1.ServiceAccount
+	// If the user has disabled rbac manager or specified comp.Spec.ServiceAccountName, it is now the user's responsibility to
+	// provide appropriate serviceaccount.
+	if serviceAccountName = transCtx.Component.Spec.ServiceAccountName; serviceAccountName != "" {
+		// if user provided serviceaccount does not exist, raise error
+		sa := &corev1.ServiceAccount{}
+		if err := transCtx.Client.Get(transCtx.Context, types.NamespacedName{Namespace: synthesizedComp.Namespace, Name: serviceAccountName}, sa); err != nil {
+			if errors.IsNotFound(err) {
+				transCtx.EventRecorder.Event(transCtx.Cluster, corev1.EventTypeWarning, EventReasonRBACManager, fmt.Sprintf("serviceaccount %v not found", serviceAccountName))
+			}
+			return err
+		}
 	}
-	if serviceAccount == nil {
-		transCtx.Logger.V(1).Info("buildServiceAccounts returns serviceAccount nil")
+	if !viper.GetBool(constant.EnableRBACManager) {
+		transCtx.EventRecorder.Event(transCtx.Cluster, corev1.EventTypeWarning, EventReasonRBACManager, "RBAC manager is disabled")
 		return nil
 	}
 
-	if isServiceAccountExist(transCtx, serviceAccount.Name) {
-		return nil
+	graphCli, _ := transCtx.Client.(model.GraphClient)
+
+	if err := cleanOldResource(transCtx, graphCli, dag, synthesizedComp); err != nil {
+		return err
 	}
 
-	if !viper.GetBool(constant.EnableRBACManager) {
-		transCtx.Logger.V(1).Info("rbac manager is disabled")
-		transCtx.EventRecorder.Event(transCtx.Cluster, corev1.EventTypeWarning,
-			string(ictrlutil.ErrorTypeNotFound), fmt.Sprintf("ServiceAccount %s is not exist", serviceAccount.Name))
-		return ictrlutil.NewRequeueError(time.Second, "RBAC manager is disabled, but service account is not exist")
+	var err error
+	if serviceAccountName == "" {
```

Comment:
Even if the service account is specified by user, should we consider setting a finalizer to prevent accidental deletion by the user?